ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1180 @@
+            this.breadcrumbsGroup = void 0 as any;
+        } else if (
+            chart.drillUpButton
+        ) {
+            breadcrumbsList.length = 0;
+            chart.drillUpButton.destroy();
+            chart.drillUpButton = void 0;
+        }
+        chart.getMargins();
+    }
+
+    /**
+     * Default button text formatter.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#getButtonText
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} breadcrumb
+     *        Breadcrumb.
+     * @return {string}
+     *         Formatted text.
+     */
+    public getButtonText(this: Breadcrumbs, breadcrumb: Array<number|null|Point|SeriesType|SVGAElement>): string {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsOptions = breadcrumbs.options,
+            lang = chart.options.lang,
+            textFormat = pick(breadcrumbsOptions.format,
+                breadcrumbsOptions.showFullPath ? '{level.name}' : '← {level.name}'
+            );
+        let returnText = breadcrumbsOptions.formatter && breadcrumbsOptions.formatter(breadcrumb, breadcrumbs) ||
+                format(textFormat, { level: (breadcrumb[1] as Point) }, chart) || '';
+
+        if (returnText === '← ' && lang && lang.mainBreadcrumb) {
+            returnText = !breadcrumbsOptions.showFullPath ? '← ' + lang.mainBreadcrumb : lang.mainBreadcrumb;
+        }
+        return returnText;
+    }
+
+    /**
+     * Perform the drillUp action.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#jumpUp
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public jumpUp(this: Breadcrumbs): void {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart;
+
+        if (this.series && this.series.drillUp) {
+            this.series.drillUp();
+        } else {
+            chart.drillUp();
+            breadcrumbs.calculateLevel();
+        }
+    }
+
+    /**
+     * Performs multiple drillups based on the provided number.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#jumpBy
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} jumpsAmount
+     *        A number of drillups that needs to be performed.
+     */
+    public jumpBy(this: Breadcrumbs, jumpsAmount: number): void {
+        const breadcrumbsList = this.breadcrumbsList;
+
+        // if defined, new level is a current level - jumps amount
+        // if not defined, new level is usually 0 - the first level in list.
+        if (breadcrumbsList.length) {
+            const jumpsNumber = defined(jumpsAmount) ?
+                breadcrumbsList[breadcrumbsList.length - 1][0] - jumpsAmount :
+                breadcrumbsList[breadcrumbsList.length - 1][0];
```

Comment:
Instead of checking `breadcrumbsList[index][0]`, could we simply use `this.level`? The same in all other places.